### ISSUE TITLE
`mem::MaybeUninit::uninit().assume_init()` is undefined behavior

### ISSUE THREAD

**Issue by MarkDDR** [gamebooster/soundboard#38]
(Copied from the issue I made on the original repo over here https://github.com/UnwrittenFun/hotkey-rs/issues/4)

There are a couple places in the code, particularly in extern/hotkey-rs, where you do something along the lines of

```rust
let mut event: SomeType = mem::MaybeUninit::uninit().assume_init();
```

This is immediately undefined behavior, regardless of the type. The scenarios I have seen this in the codebase are in the context of a C-style return parameter, so the fix is rather simple.

```rust
let mut event: MaybeUninit<SomeType> = mem::MaybeUninit::uninit();
// Maybe attempt to handle a potential int-code error this function may return?
get_event(event.as_mut_ptr());
let event: SomeType = event.assume_init();
```

**Comment by koutoftimer** [comment]
@MarkDDR you are absolutely right for general case:

from [`assume_init`](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#method.assume_init)
> Extracts the value from the `MaybeUninit<T>` container. This is a great way to ensure that the data will get dropped, because the resulting `T` is subject to the usual drop handling.

from [`uninit`](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#method.uninit)
> Creates a new `MaybeUninit<T>` in an uninitialized state.
>
> Note that dropping a `MaybeUninit<T>` will never call `T`'s drop code. It is your responsibility to make sure `T` gets dropped if it got initialized.

`GetMessageW` does not expect chuck of memory, it has pointer to, to be in defined state as in case of Rust. It can be allocated with `malloc` or any other fucntions that just gives you memory with state from previous usage. It is `GetMessageW` responsibility to write each and every field and ensure that event struct gets initialized right.

[`winapi::um::winuser::MSG`](https://docs.rs/winapi/0.3.9/winapi/um/winuser/struct.MSG.html) has no pointers in it. If you will drop `MaybeUninit<T>` and `MSG` result will be the same.

So what the difference between your solution and author's? - Yours have one more line of code.

Or, in other words, can you supply cases when it is really undefined behavior? According to current code base.

**Comment by skomski** [comment]
Fix cc3560e55caaf427762e8a9cf3340e7bf641eb50
